Partial-CI-Tests-Structure.md

```diff
@@ -0,0 +1,39 @@
+* The script at `scripts/check_ci_test_suites_to_run.py` is used to check the CI test suites that need to run in a given PR. It does this by checking the `git diff` between the PR `HEAD` branch and the branch that is being merged into. It uses the above "root-files generator" and "test-modules mappings" to determine which tests to run and outputs it to the github workflow.
+
+The process of determining which tests to run goes like this:
+
+1. Generate a root files mapping which maps files to their root files. This file is generated at the start of every PR check run and is stored at `root-files-mapping.json`.
+2. Run the script, `scripts/check_ci_test_suites_to_run.py` inside the Github workflow which will map changed files in a PR to their root files using the mapping above. Afterwards, map those root files to tests by using the golden files containing the page modules that a test uses.
+
+# Test to Modules Mappings
+
+The test-to-modules mappings at `core/tests/test-modules-mappings` contain the "golden" mappings of a test and the Angular page modules that they use. This golden file is validated everytime a test is run and if there is a discrepancy between the generated test-to-modules mapping and the golden file mapping, then it will output an error with the Angular page modules that are extra/missing. This ensures that the "golden" mappings are kept up to date whenever a test is created or modified.
+If you get an error about the test-to-modules mapping, please do the following:
+
+* If the error only happens on a few tests, please simply follow the error message and change the golden file with the Angular page modules that the error message shows.
+* If the error happens on many tests, please consider adding it to the `COMMON_MODULES_TO_EXCLUDE` constant at `core/tests/test-dependencies/test-to-modules-matcher.ts`, with a corresponding glob of the CUJ that will cover that module, or a TODO if that CUJ doesn't exist yet. 
+
+# Root Files Config
+
+The root files config at `core/tests/root-files-config.json` contains all the valid root files that are generated in the root files mapping. A root file is a file in the dependency graph which has no parents, the most common root files in the codebase are page modules and other files like `README.md`. The "root files mapping generator" validates the `root-files-config.json` by comparing the PR's root files in the mapping with the root files found in the config, if there are any discrepancies, the generator will error. This ensures that the "root-files config" is always up to date with every PR.
+If you get an error that there is a root file not contained in these lists, please add them to one of these lists depending on the context of the file:
```

Comment:
Put another blank line above this, it can be a separate paragraph.

Reply:
Done.